enable OP steer with set button on non OP long

xps-genesis opened the pull request:
For non OP long vehicles-(stock SCC or normal cruise)
OP steer should enable/disable with cruise on/off button
OP steer should not cancel with cruise "cancel" button( this is only recommended for OP long vehicles)

xps-genesis committed:
Avoid OP steer cancel with cancel button on non OP long

For non OP long vehicles-
OP steer should be enable/disable with acc on/off button
OP steer should not cancel with cruise cancel button( this is only required for OP long vehicles)

xx979xx commented:
@xps-genesis but what about if user wants to disable OP steer without turn off curies control complacently and lose stored set speed? (cruise control has to disable and enabled again to re-enabled OP though)
what do you think?

xps-genesis commented:
according to me we should enable OP steer with any of these conditions for non OP long vehicles
1.) cruise set-/res+ and cruise button on
2.) leading edge of cruise button on

OP steer should disable with any of these conditions
1.) brake/gas/gear not in D/seatbelt/door etc..
2.) cruise button off

so when driver 1st presses cruise button on, OP steer engages. he is able to cancel OP steer by pressing gas/brake or pressing cruise button off.
when OP steer is engaged, driver cancels OP steer with gas/brake, he is able to engage op steer with set-/res+ buttons.

xps-genesis committed:
update to new description

according to me we should enable OP steer with any of these conditions
1.) cruise set-/res+
2.) cruise button on

OP steer should disable with any of these conditions
1.) brake/gas
2.) cruise button off

xps-genesis committed:
Update interface.py

xps-genesis committed:
added enabled condition to set-/res+

xx979xx commented:
@xps-genesis that make sense (except disable on pedal which I hate), but I do not see why cruise cancel is not an option.

I think it is a user preference, and UI toggle should  do it

personally I like OP steer only to be enabled by ( 1- cruise on 2- cruise set-)
and disabled by (1- cruise off 2- cruise cancel)
this give the user all options by using buttons only, use OP steer with cruise enabled/ without cruise enabled (cancel then resume )

xps-genesis committed:
updated to new req

now the OP steer can be enabled by ( 1- cruise on 2- cruise set- 3- cruise res+)
and disabled by (1- cruise off 2- cruise cancel)

xx979xx committed:
enable on cruise set
